apis/eks/v1beta1/types.go

```diff
@@ -0,0 +1,313 @@
+	LogTypeScheduler         LogType = "scheduler"
+)
+
+// ClusterParameters define the desired state of an AWS Elastic Kubernetes
+// Service cluster.
+type ClusterParameters struct {
+	// The encryption configuration for the cluster.
+	// +immutable
+	// +optional
+	EncryptionConfig []EncryptionConfig `json:"encryptionConfig,omitempty"`
+
+	// Enable or disable exporting the Kubernetes control plane logs for your cluster
+	// to CloudWatch Logs. By default, cluster control plane logs aren't exported
+	// to CloudWatch Logs. For more information, see Amazon EKS Cluster Control
+	// Plane Logs (https://docs.aws.amazon.com/eks/latest/userguide/control-plane-logs.html)
+	// in the Amazon EKS User Guide .
+	//
+	// CloudWatch Logs ingestion, archive storage, and data scanning rates apply
+	// to exported control plane logs. For more information, see Amazon CloudWatch
+	// Pricing (http://aws.amazon.com/cloudwatch/pricing/).
+	// +optional
+	Logging *Logging `json:"logging,omitempty"`
+
+	// The VPC configuration used by the cluster control plane. Amazon EKS VPC resources
+	// have specific requirements to work properly with Kubernetes. For more information,
+	// see Cluster VPC Considerations (https://docs.aws.amazon.com/eks/latest/userguide/network_reqs.html)
+	// and Cluster Security Group Considerations (https://docs.aws.amazon.com/eks/latest/userguide/sec-group-reqs.html)
+	// in the Amazon EKS User Guide. You must specify at least two subnets. You
+	// can specify up to five security groups, but we recommend that you use a dedicated
+	// security group for your cluster control plane.
+	//
+	// ResourcesVpcConfig is a required field
+	ResourcesVpcConfig VpcConfigRequest `json:"resourcesVpcConfig"`
+
+	// The Amazon Resource Name (ARN) of the IAM role that provides permissions
+	// for Amazon EKS to make calls to other AWS API operations on your behalf.
+	// For more information, see Amazon EKS Service IAM Role (https://docs.aws.amazon.com/eks/latest/userguide/service_IAM_role.html)
+	// in the Amazon EKS User Guide .
+	//
+	// RoleArn is a required field
+	// +immutable
+	// +optional
+	RoleArn *string `json:"roleArn"`
+
+	// RoleArnRef is a reference to an IAMRole used to set
+	// the RoleArn.
+	// +immutable
+	// +optional
+	RoleArnRef *runtimev1alpha1.Reference `json:"roleArnRef,omitempty"`
+
+	// RoleArnSelector selects references to IAMRole used
+	// to set the RoleArn.
+	// +immutable
+	// +optional
+	RoleArnSelector *runtimev1alpha1.Selector `json:"roleArnSelector,omitempty"`
+
+	// The metadata to apply to the cluster to assist with categorization and organization.
+	// Each tag consists of a key and an optional value, both of which you define.
+	// +optional
+	Tags map[string]string `json:"tags,omitempty"`
+
+	// The desired Kubernetes version for your cluster. If you don't specify a value
+	// here, the latest version available in Amazon EKS is used.
+	// +optional
+	Version *string `json:"version,omitempty"`
```

Comment:
Is there a format specification for this field? It'd be easier for people to know what's expected as version.

Reply:
@muvaf this actually just takes `1.15`, `1.16`, etc. and the EKS platform version is a separate field in the status. I have tested providing this manually and updating it to new version. I am not sure we should impose a restriction here as AWS does not specifically give one and it would increase maintenance burden if we had to keep a list of valid values here.

Reply:
I was just wondering whether we should include a format in the comment like it needs to be `x.yy` format rather than `x.yy.zz` or actual eks version like `x.yy.zz.eks99`. If it's just plain `x.yy`, maybe add a word like `Example: 1.15`?

Reply:
I like the idea of the example, added :+1: